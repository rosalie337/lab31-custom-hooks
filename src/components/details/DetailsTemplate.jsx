/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';

function DetailsTemplate({ name, image, allies, enemies }) {

  const allyElements = allies?.map((ally, i) => {
    return <li key={i} style={{ fontWeight: 'normal' }}>{ally}</li>;
  });
        
  const enemyElements = enemies?.map((enemy, i) => {
    return <li key={(i * Math.random()) } style={{ fontWeight: 'normal' }}>{enemy}</li>;
  });

  return (
    <>
      <figure>
        <img src={image} />
        <figcaption data-testid="name" style={{ fontSize: '25px' }}>{name}</figcaption>
      </figure>
      
      { (allyElements?.length > 0)
        && <ul style={{ fontWeight: 'bold' }}>Allies: {allyElements} </ul>
      }

      { (enemyElements?.length > 0)
        && <ul style={{ fontWeight: 'bold' }}>Enemies: {enemyElements} </ul>
      }
    </>
  );
}

DetailsTemplate.propTypes = {
  name: PropTypes.string.isRequired,
  photoUrl: PropTypes.string.isRequired,
  allies: PropTypes.array,
  enemies: PropTypes.array,
};

export default DetailsTemplate;
