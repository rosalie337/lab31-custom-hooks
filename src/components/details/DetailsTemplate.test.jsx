/* eslint-disable max-len */
import React from 'react';
import { render, cleanup, waitFor } from '@testing-library/react';
import DetailsTemplate from './DetailsTemplate';

describe('DetailsTemplate component', () => {
  afterEach(() => cleanup());
  it('renders DetailsTemplate', () => {
    const { asFragment } = render(<DetailsTemplate
      name="Baatar Jr."
      photoUrl="https://vignette.wikia.nocookie.net/avatar/images/4/45/Baatar_Jr..png/revision/latest?cb=20150409193510"
    />);    

    return waitFor(() => {expect(asFragment()).toMatchSnapshot()});
  });
});