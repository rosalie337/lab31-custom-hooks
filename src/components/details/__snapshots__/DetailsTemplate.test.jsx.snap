// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsTemplate component renders DetailsTemplate 1`] = `
<DocumentFragment>
  <figure>
    <img
      src="https://vignette.wikia.nocookie.net/avatar/images/4/45/Baatar_Jr..png/revision/latest?cb=20150409193510"
    />
    <figcaption
      data-testid="name"
      style="font-size: 25px;"
    >
      Baatar Jr.
    </figcaption>
  </figure>
</DocumentFragment>
`;
